website/blog/2022-03-04-modifying-blocks.md

```diff
@@ -0,0 +1,275 @@
+<!--truncate-->
+
+Since Eightshift Development kit is a starter theme, made for developers to jumpstart and speed up their development, you're welcome to modify files directly in it. There is no need to create a child theme to protect it from updates.
+
+### Modifying a block or a component?
+Since we used Quote block in our previous post, we will continue using it as an example since it's fairly simple. One of the first questions you may ask yourself could be: "Should I modify a component or a block?". And the answer is - it depends.
+
+If you compare **_components/quote/quote.php_** (component) and **_custom/quote/quote.php_** (block), you'll notice most HTML code is inside a component, while block pretty much only renders the Quote component inside a wrapper. This example will walk you through the whole process of adding a new attribute and its options to a block.
+
+### What will we do?
+
+We want to add one new feature to this block. An option to choose a color theme for the Quote block. These are the specs:
+- background with rounded corners
+- three color themes: blue, green, yellow
+- the background has a lighter shade of the selected color
+- quote icon has a stronger shade of the selected color
+
+### Adding background
+
+Initial background styling is fairly straightforward. Navigate to **_src/Blocks/components/quote/quote-style.scss_** and paste the following code inside `.quote` class:
+```css
+padding: calc(var(--base-font-size) * 1.6rem);
+border-radius: calc(var(--base-font-size) * 2rem);
+background-color: global-settings(colors, light);
+```
+You may notice we're using `calc` instead of directly writing values in rems. This way makes it much easier to calculate pixel size since `calc(var(--base-font-size) * 1.6rem)` equals `16px`.
+
+For now, we will add a simple light grey background to see how it looks. We'll replace this value later with a CSS variable.
+>**Note:** don't hardcode hex color values directly inside your component. Instead, use colors defined in your global manifest.
+
+You'll notice that the changes are visible both in the editor and on the frontend. Since Gutenberg editor has some specific quirks, sometimes you will need to add additional styling only for the editor. In case we need to override something in the editor for our Quote component, we would simply create **_quote-editor.scss_**.
+
+### Adding new colors to your project
+
+Since the theme currently doesn't have all the colors we need to implement the color theme feature for our block, we have to add a few new colors to our project. We will use the colors already defined in the manifest for icon color, but we need to add lighter variations of those colors to use them for the background. Navigate to your global manifest, which is located inside **_src/Blocks/manifest.json_** and add the following values inside `colors`:
```

Comment:
```suggestion
Because the theme currently doesn't have all the colors, we need to implement the color theme feature for our block. We will use the colors already defined in the manifest for icon color, but we need to add lighter variations of those colors to use them for the background. Navigate to your global manifest, which is located inside **_src/Blocks/manifest.json_** and add the following values inside `colors`:
```